components/local-app/pkg/auth/auth.go

```diff
@@ -120,26 +120,25 @@ const html = `
 </html>`
 
 // NOTE: the port ranges all need to be valid redirect URI's in the backend
-const STARTING_PORT_NUM = 63110
-const ENDING_PORT_NUM = 63120
+const (
+	StartingPortNum = 63110
+	EndingPortNum   = 63120
+)
 
 // Login walks through the login flow for obtaining a Gitpod token
 func Login(ctx context.Context, opts LoginOpts) (token string, err error) {
 	// Try a range of ports for local redirect server
-	var rl net.Listener
-	var port int
-	for port = STARTING_PORT_NUM; port < ENDING_PORT_NUM || rl == nil; port++ {
-		rl, err = net.Listen("tcp4", fmt.Sprintf("127.0.0.1:%d", port))
-		if err != nil {
-			logrus.Infof("Could not open port:%d got error:%s\nTrying next port\n", port, err)
-			continue
-		}
-		defer rl.Close()
-	}
-	if rl == nil {
-		return "", xerrors.Errorf("could not open any valid port in range %d - %d", STARTING_PORT_NUM, ENDING_PORT_NUM)
+	rl, port, err := findOpenPortInRange(StartingPortNum, EndingPortNum)
+	if err != nil {
+		return "", err
 	}
 
+	defer func() {
+		if closeErr := rl.Close(); closeErr != nil {
+			logrus.WithField("port", port).WithError(closeErr).Warn("Failed to to close listener")
```

Comment:
> WARN[0005] Failed to to close listener                   error="close tcp4 127.0.0.1:63110: use of closed network connection" port=63110

After test, we can see this warning every time. we don't need this warning, because it will always appear if auth works(it will close by [this one](https://github.com/gitpod-io/gitpod/pull/8956/files#diff-45dc697108fc2017a12d5505d0cfe7279372d64248c62238a15d1c4ade7be7f7R166) too).